Move Blog, Newsroom, Activity log to Wagtail

kurtw opened the pull request:
I've combined this into a single PR because you need each piece to test it all without breaking things.

## Additions

- Legacy page types to migrate old content from WordPress
- import-data processors to do the migrations
- corresponding templates
- rss feeds
- new references to Activity log in the category reference
- Reports is now a thing in Activity Log

## Removals

- i removed the old stuff that we aren't going to use anymore, like urls and template files

## Changes

- The `get_page_set` function to get the pages for each of the filterable lists ended up being so different that i ended up breaking into its corresponding page class to separate concerns.
- legacy content uses the same article.html file it did before but i changed it to be wagtail page compatible.
- view more works with wagtail pages now
- bug fixing some stuff
- some other stuff i probably forgot

## Testing

- get the latest from refresh
- make sure you have the database option in local.py set to run migrations that create foreign key references
- run the migrations
- under the home page, add an Activity Log Page
 - make sure there's a filterable list with categories selected, and **Activity Log** selected as the page type
 - ![screen shot 2016-04-19 at 10 19 47 am](https://cloud.githubusercontent.com/assets/254877/14642382/46b8d764-0618-11e6-8378-28de8ec78f41.png)

- under the about us section, add a **Newsroom Landing Page** and a Blog **Sublanding filterable Page** 
 - again, make sure that there's a filterable list, and the page types are selected (Newsroom for Newsroom, Blog for blog)
- run `./cfgov/manage.py import_data posts legacyblogpage --parent blog -u user -p pass`
- run `./cfgov/manage.py import_data newsroom legacynewsroompage --parent newsroom -u user -p pass`
 - *Note*: a couple errors are fine, just let me know what they are. there are some blog posts that should exist in production right now
- Play around with the list filters on each page to make sure that they work. things like filtering on date, topic, multiple categories, and the like.
- Next, go to a Blog post and make sure that the further reading organism is pulling content if it exists for that tag, and that the View More link goes to the activity log with the appropriate tags selected.
- I think that's it....

## Review

- @rosskarchner 
- @kave 
- @richaagarwal 

## Screenshots
Blog Landing Page
![screen shot 2016-04-18 at 7 20 09 pm](https://cloud.githubusercontent.com/assets/1412978/14622796/972bad3a-059a-11e6-8d40-c34f14955268.png)


LegacyBlogPage
![screen shot 2016-04-18 at 7 16 07 pm](https://cloud.githubusercontent.com/assets/1412978/14622711/0c3260e8-059a-11e6-9227-99f82040c5a1.png)

Newsroom Landing Page (note the side nav. this is pulling in all browse type pages but newsroom is currently not doing this. its a flag in the admin to set it right. Also, theres a child page of Newsroom called press sources that isn't there; more below.)
![screen shot 2016-04-18 at 7 17 13 pm](https://cloud.githubusercontent.com/assets/1412978/14622781/83ec44e6-059a-11e6-9cdd-0576867cf02e.png)


LegacyNewsroomPage (same template as blog but this doesn't have further reading content for lack of tags)
![screen shot 2016-04-18 at 7 17 48 pm](https://cloud.githubusercontent.com/assets/1412978/14622749/407902b2-059a-11e6-8ff0-2425074a26d5.png)

Activity Log pulling in reports alongside blog
![screen shot 2016-04-18 at 7 19 00 pm](https://cloud.githubusercontent.com/assets/1412978/14622771/75429b5c-059a-11e6-95c1-40a9a5672460.png)


## Notes

- @schaferjh Can the Press Sources page (under Newsroom) be a Wagtail page right now?

## Todos

- We still need to make the new BlogPage and the new NewsroomPage classes and templates
- Possibly create the Press Sources page

kave commented:
good work on getting this done, i'll try this out in the morning

kave commented:
migrations conflicts

kave commented:
For blog the related posts "Further Reading" organism isn't rendering when added to the page

kave commented:
on this [page](http://localhost:8000/about-us/blog/new-signs-of-trouble-for-student-loan-borrowers) clicking the authors name doesn't redirect to the parent filterable page. It just reloads the page.

kave commented on cfgov/v1/forms.py:
```diff
@@ -314,3 +317,53 @@ def get_query_strings(self):
+            blog = CFGOVPage.objects.get(slug='blog')
+            blogs = CFGOVPage.objects.live_shared(hostname).descendant_of(blog)
+        except CFGOVPage.DoesNotExist:
+            print 'A blog landing page needs to be made'
```
lets change this is a raised exception with this message

kurtw replied:
I don't want to prevent the page from being viewable if only one portion of content doesn't exist. Is there a better way to show that it needs to be created without totally erroring out?

kave commented on cfgov/v1/forms.py:
```diff
@@ -314,3 +317,53 @@ def get_query_strings(self):
+
+class ActivityLogFilterForm(NewsroomFilterForm):
+    def per_page_limit(self):
+        return 100
```
this could be useful as a global env variable

kurtw replied:
Yeah, but unless the need arises, I'd rather keep things the way they are. Up to be overruled @rosskarchner @richaagarwal.

kave commented on cfgov/v1/forms.py:
```diff
@@ -314,3 +317,53 @@ def get_query_strings(self):
+                page = CFGOVPage.objects.get(slug=slug)
+                pages[slug] = CFGOVPage.objects.live_shared(hostname).descendant_of(page)
+            except CFGOVPage.DoesNotExist:
+                print slug, 'does not exist'
```
when would a slug not exist?

kurtw replied:
If there aren't pages that have the slugs `blog`, `newsroom`, or `research-reports`.

kave commented on cfgov/v1/models/browse_filterable_page.py:
```diff
@@ -67,3 +68,30 @@ def get_form_class(self):
+                blogs = AbstractFilterPage.objects.live_shared(hostname).descendant_of(blog).filter(form.generate_query())
+                filter_pages = list(chain(filter_pages, blogs))
+            except base.CFGOVPage.DoesNotExist:
+                print 'Blog does not exist'
```
same here raised exception

kurtw replied:
https://github.com/cfpb/cfgov-refresh/pull/1922#discussion_r60246792

kave commented on cfgov/v1/models/sublanding_filterable_page.py:
```diff
@@ -50,9 +51,49 @@ def get_context(self, request, *args, **kwargs):
+                parent = CFGOVPage.objects.get(slug='newsroom')
+                results['newsroom'] = AbstractFilterPage.objects.live_shared(hostname).descendant_of(parent).filter(form.generate_query())
+            except CFGOVPage.DoesNotExist:
+                print 'Newsroom does not exist'
```
raised exception

kave replied:
guaranteed in the future we'll need to add validation for the fields on the wagtail admin.

kurtw replied:
> guaranteed in the future we'll need to add validation for the fields on the wagtail admin.
 yup

> raised exception, so we'll be email notified.
As per [my comment](https://github.com/cfpb/cfgov-refresh/pull/1922#discussion_r60246792), being notified by email makes sense, but again I don't want to prevent the whole page from being viewable just because one source of content is dead. Is there a better way?

kave commented on cfgov/v1/util/ref.py:
```diff
@@ -33,7 +34,7 @@
     ('newsroom', 'Newsroom'),
     ('notice-opportunity-comment', 'Notice and Opportunity for Comment'),
-    ('research-report', 'Research Report'),
+    ('research-reports', 'Research Report'),
```
would you need to write a script to update legacy pages?

kurtw replied:
Only one page has this value, so I'm just going to change it manually.

kave commented on cfgov/v1/util/util.py:
```diff
@@ -85,7 +85,7 @@ def get_form_id(page, get_request):
 def instanceOfBrowseOrFilterablePages(page):
     from ..models import BrowsePage, BrowseFilterablePage
-    return isinstance(page, BrowsePage) or isinstance(page, BrowseFilterablePage)
+    return isinstance(page, (BrowsePage, BrowseFilterablePage))
```
oh cool didn't know you could do that

kave commented:
added a few comments, but after that 👍  everything looks good

rosskarchner commented:
This is pretty cool, @kurtw 

One thing I noticed, is the the view-more link seems the be relative to the post, so on a Mortgage post, the URL is:

http://localhost:8000/about-us/blog/mortgage-moves-how-much-can-you-afford/#?filter0_topics=Mortgages


Which seems wrong.

rosskarchner commented:
Here's something weird, on a completely unrelated page--

TypeError at /policy-compliance/notice-opportunities-comment/
macro 'render' takes no keyword argument 'post_date_description'

switching back to the 'flapjack' branch resolves it, so it's definitely code introduced here.

kave commented:
yea for the ` /policy-compliance/notice-opportunities-comment/` there was an incorrect merge conflict https://github.com/cfpb/cfgov-refresh/pull/1922/files#diff-e64c1122508cbcf9a1de90e7879c62f9L46 

`, post_date_description=''` should be added back in

rosskarchner commented:
Do I have to do anything special to make the title show up on the Newsroom Landing Page?

local on the left, beta on the right

![screen shot 2016-04-19 at 1 50 19 pm](https://cloud.githubusercontent.com/assets/235397/14649587/c59f6f08-0635-11e6-84c5-122c275c0dc9.png)

rosskarchner commented:
Last thing, I'm ppppprrrreeettttyyyy sure I followed the instructions this time, but newsroom stuff isn't showing up in the activity log.

anselmbradford commented on cfgov/jinja2/v1/blog/legacy_blog_page.html:
```diff
@@ -16,6 +16,11 @@
     {% include 'article.html' %}
+    <div class="block">
+        <p>
+            Due to technical issues, the commenting feature of our blog is temporarily unavailable. We’re working to bring this functionality back, and look forward to hearing your feedback and comments about the CFPB’s work soon.
```
Trivial: break to new lines after periods so it's easier to read in GH.

schaferjh commented:
Just to follow up publicly on the comment left for me in Notes:

> @schaferjh Can the Press Sources page (under Newsroom) be a Wagtail page right now?

Not right now. But moving it and other hard-coded V1/Refresh pages into Wagtail ASAP will be a post-launch goal!

kurtw commented:
Fixed the post preview issue, the About us and Careers landing pages Latest Activities issue, and changed the the Legacy page models to use a Streamfield and an instance of RawHTMLBlock to pull in content so there isn’t the chance of accidentally ripping out content in Wagtail’s RichTextField.

kurtw commented:
@rosskarchner I fixed the Activity Log. Pull down and try now.

richaagarwal commented:
👍 assuming the categories filter is inclusive

rosskarchner commented:
@kurtw is there a chance the RSS feed is still pulling from Sheer(like)? It doesn't include the latest blog post:   http://www.consumerfinance.gov/feed/blog/
